### ISSUE TITLE
Error when running `heroku fork` on Linux: Error: Cannot find module 'co'

### ISSUE THREAD

**Issue by Taytay** [heroku/heroku#1571]
It sounds like this might be due to an unfinished update on one of our 4 build machines?
Perhaps related to GH-1567

```
00:18:10.907 module.js:339
00:18:10.907     throw err;
00:18:10.907           ^
00:18:10.908 Error: Cannot find module 'co'
00:18:10.908     at Function.Module._resolveFilename (module.js:337:15)
00:18:10.908     at Function.Module._load (module.js:287:25)
00:18:10.908     at Module.require (module.js:366:17)
00:18:10.908     at require (module.js:385:17)
00:18:10.909     at Object.<anonymous> (/home/vagrant/.heroku/node_modules/heroku-fork/commands/fork.js:3:18)
00:18:10.909     at Module._compile (module.js:431:26)
00:18:10.909     at Object.Module._extensions..js (module.js:449:10)
00:18:10.909     at Module.load (module.js:356:32)
00:18:10.909     at Function.Module._load (module.js:311:12)
00:18:10.910     at Module.require (module.js:366:17)
```

On Mac, I can run `heroku update` to wait for everything to be updated, right?
On Linux though, it looks like running `heroku update` will not block and wait for the CLI and plugins to update. Is there a way for me to block and wait for everything to be updated? That way, I could do it at the beginning of our build, to decrease the chance of our build machine discovering that it's only partially updated midway through. 

Or perhaps your fixes in GH-1567 will make that request moot?

**Comment by Taytay** [comment]
Running it again didn't help (unlike GH-1567, where that just made it go away), so I have logged in and done this:
```
$ heroku --version
heroku-toolbelt/3.36.8 (x86_64-linux) ruby/2.2.2
heroku-cli/4.19.19-6dfff81 (amd64-linux) go1.4.2
ERROR: json: cannot unmarshal array into Go value of type map[string]gode.Package
npm ERR! max depth reached: co@^4.5.4, required by heroku-fork@3.0.3
npm ERR! max depth reached: heroku-cli-util@^3.6.0, required by heroku-fork@3.0.3
npm ERR! max depth reached: lodash@^3.2.0, required by heroku-fork@3.0.3
=== Installed Plugins
heroku-pipeline
```

More info to come as I learn more about this...

**Comment by Taytay** [comment]
Other linux machines we have are fine though: 
```
heroku version
heroku-toolbelt/3.36.7 (x86_64-linux) ruby/2.2.2
heroku-cli/4.19.19-6dfff81 (amd64-linux) go1.4.2
=== Installed Plugins
heroku-apps@0.1.4
heroku-fork@3.0.3
heroku-pipeline
```

So, this might just be an error on this particular machine.

**Comment by Taytay** [comment]
Removed and reinstalled Heroku Toolbelt, and now `fork` works again:
```
$ heroku version
heroku-toolbelt/3.36.8 (x86_64-linux) ruby/2.2.2
heroku-cli/4.19.19-6dfff81 (amd64-linux) go1.4.2
=== Installed Plugins
heroku-fork@3.0.3
```
That's the oddest thing. Feel free to close out, since it's a transient error. I wanted to leave it open just in case you or others happen to run into this.

**Comment by dickeyxxx** [comment]
I was able to identify a few bugs that I resolved in 4.19.20. One includes the locking around an individual plugin, so hopefully that will make the `Error: cannot find module` bug a thing of the past. However, maybe not until everything goes through one more round of updates.

There was also a bug with the update command that it was using the autoupdate file to check if it needed to update. That's the bug you saw, that calling `heroku update` didn't do an update, it just checked the autoupdate file and saw it didn't need to update.
